### ISSUE TITLE
[Sentry] Known performance and usability issues

### ISSUE THREAD

**Issue by jhouse-solvd** [department-of-veterans-affairs/va.gov-team#27032]
## Description
Sentry is currently experiencing performance issues that make it sporadically unusable or sub-optimal. Ops is working to take any immediate remediation steps possible, while simultaneously working w/ Platform leadership and stakeholders to discuss alternative solutions. 

## Background/context/resources
Sentry was recently upgraded and migrated to EKS as part of the large infrastructure recommit initiative. 
The self-hosted solution lacks quality, decisive support. There seems to be a push towards adoption of Sentry's cloud-hosted offering, but the platform may be unable to use it due to ATO requirements (ie sensitive data; external provider currently lacking FedRamp designation)

## Technical notes / issues

- Rolling back to the legacy is not an option. 
- The ingestion of events is causing sporadic issues (related to events volume/retention/frequency). 
- Login / UI are sporadically extremely slow to the point of being unusable.
- Additional issues/tasks [HERE](https://github.com/department-of-veterans-affairs/devops/projects/19)

---
## Tasks
- [x] See [HERE](https://github.com/department-of-veterans-affairs/devops/projects/19) 

## Definition of Done
- [x] We will identify a path forward for managing/providing a usable error capture tool.

**Comment by mchelen-gov** [comment]
Related work that may help:
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/27231
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/25271
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/24144

**Comment by mleclerc00** [comment]
Short writeup of some findings
https://github.com/department-of-veterans-affairs/devops/issues/9580#issuecomment-881088834

**Comment by jhouse-solvd** [comment]
Some more granular tasks being tracked here: https://github.com/department-of-veterans-affairs/devops/projects/19

Will post to ticket body above as well.

**Comment by jhouse-solvd** [comment]
A great summary from Michael Fleet regarding technical details, status, plan, and actions underway: 

"Hey folks! We are aware that Sentry is not behaving as expected. We are working on it. We are also pretty much between a rock and hard place when it comes to fixing it.
TL;DR: We are doing the best that we can. We are not ignoring Sentry at all.
At the risk of over-sharing, here are the plot highlights:

We upgraded to Sentry 10, then 20, in order to access features that we have all been clamoring for. And in doing so, we found that Sentry has significantly complicated its “self-hosted” architecture. What was once 3 processes running in 1 container is now 3+ processes each in roughly a dozen containers. It’s complicated to troubleshoot. It’s poorly documented. And Sentry folks shrug at self-hosted support requests. Yes, we’ve considered rolling back, and it’s not worth the time to do so.
Plan A is to push forward with Sentry 20, debug the new gotchas, via give-and-take with the self-hosted Sentry community, and hope that gets us back to healthy-ish. We are actively working on this plan right now.

Plan B is migrate away from Sentry. Its clear Sentry has no desire to support self-hosted customers anymore. If we can’t support ourselves, Plan B may be our only, if quite disruptive option.

I know that Plan B will evoke some strong feelings, so let’s table it for now. It’s not worth getting riled up about if Plan A works."

**Comment by jhouse-solvd** [comment]
Update: 

- We did some [cleanup in the database](https://app.zenhub.com/workspace/o/department-of-veterans-affairs/va.gov-team/issues/28615) that took us from ~35M discrete environments to ~14 (continuing to trim)
- Feedback from at least one power user (@annaswims) says we can tentatively say that this is in a usable state

Notes: 

- Check Sentry first if you haven't in a while
- Please chat with us in the [#vsp-operations slack channel](https://dsva.slack.com/archives/CJYRZK2HH), during in Ops office hours on M/W/F @ 1300 ET, or by opening a support issue for specific problems that you or your team may be experiencing.
- It will be helpful if you can be as detailed and specific as possible (including how to reproduce) to help Ops support personnel more quickly narrow down what you may be experiencing., in a while

**Comment by jhouse-solvd** [comment]
As far as we know, Sentry is stable and usable at this time. 

We are continuing to add add'l observability, monitoring, and alerting as needed to stay on top of any issues that may or could arise. 

Please open specific support issues as needed moving forward. 

Closing.
